staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -188,19 +152,16 @@ func WithAuditID(ctx context.Context, auditID types.UID) {
 		return
 	}
 	ac := AuditContextFrom(ctx)
-	if ac == nil {
+	if !ac.Enabled() {
 		return
 	}
-	ac.auditID = auditID
-	if ac.Event != nil {
-		ac.Event.AuditID = auditID
-	}
+	ac.Event.AuditID = auditID
 }
 
 // AuditIDFrom returns the value of the audit ID from the request context.
 func AuditIDFrom(ctx context.Context) (types.UID, bool) {
 	if ac := AuditContextFrom(ctx); ac != nil {
-		return ac.auditID, ac.auditID != ""
+		return ac.Event.AuditID, true
```

Comment:
Nit: Is it always initialized with an ID, so that we don't need to check for empty string as it initialized by `WithAuditInit`. Isn't it making it easier to make a mistake if the audit pkg is used outside of `k8s.io/apiserver`? I could imagine that it takes some debugging to figure out that it can be true on an empty string.

Reply:
Yes, the audit ID is always initialized at the same time as the audit context: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit_init.go#L41-L50

But you bring up a good point. Looking at everywhere this is used, I think the bool return value is really meant to indicate whether auditing is enabled. This method isn't used in very many places, so I think we should probably just inline it with an explicit `auditContext.Enabled()` check instead.

Reply:
Actually, since the eventual goal is to make the `AuditContext.Event` non-exported, I think it makes sense to keep this function as-is for now.

Reply:
According to the updated description, `ac.Enabled()` should be returned instead, in case the policy level is "None".
I am not sure whether that would still align with the original purpose of the bool.

Reply:
Is this comment still open?